### PR TITLE
Zipf's Law updates to configuration chapter

### PR DESCRIPTION
**Pull request by DamienIrving**

### CODE REVIEW

**Comment by DamienIrving** [comment]
This one is ready for review.

@gvwilson - Did you have anything else (i.e. beyond what I've included) in mind for the configuration chapter?

File: py-rse/configuration.Rmd [hunk 1]
```diff
@@ -51,162 +53,361 @@ fonts:
...
+we could use any of the overlay options described above:
+
+-  Edit the system-wide matplotlib configuration YAML file known as the `matplotlibrc` file
+-  Create a user-specific matplotlib style sheet (which also uses the YAML format) 
+-  Create a job-specific YAML file that the `plotcounts.py` script can parse
+-  Add some new command-line options to `plotcounts.py` 
+
+Let's consider these options one at a time.
+
+### Edit the matplotlibrc file
```
Comment (gvwilson) [inline py-rse/configuration.Rmd]: maybe mentioned that `rc` stood (once upon a very long time ago) for "resource control"?

File: py-rse/configuration.Rmd [hunk 2]
```diff
@@ -51,162 +53,361 @@ fonts:
...
 
-DEFAULT_CONFIG = {
-    'source': 'DEFAULT',
-    'logfile': '/tmp/log.txt',
-    'quiet': False,
-    'overwrite': False,
-    'fonts': ['Verdana', 'Serif']
-}
+```text
+/Users/amira/opt/anaconda3/lib/python3.7/site-packages/matplotlib/mpl-data/matplotlibrc
```
Comment (gvwilson) [inline py-rse/configuration.Rmd]: Worth a note to deconstruct this path: the first part is where Anaconda is installed, `site-packages` is where packages go, and the rest is matplotlib's.

File: py-rse/configuration.Rmd [hunk 3]
```diff
@@ -51,162 +53,361 @@ fonts:
...
+```
 
-    if args.infile:
-        config['infile'] = args.infile
+We could uncomment those lines, change the sizes,
 
-    if args.outfile:
-        config['outfile'] = args.outfile
+```yaml
+axes.labelsize      : x-large  ## fontsize of the x any y labels
```
Comment (gvwilson) [inline py-rse/configuration.Rmd]: indent one space to line up?

File: py-rse/configuration.Rmd [hunk 4]
```diff
@@ -51,162 +53,361 @@ fonts:
...
+print(plt.style.available)
 ```
 
-```shell
-$ python config_template.py --infile chapter2.txt --configfile thesis.yml 
+```text
+['seaborn-dark', 'seaborn-darkgrid', 'seaborn-ticks', 'fivethirtyeight', 'seaborn-whitegrid', 'classic', '_classic_test', 'fast', 'seaborn-talk', 'seaborn-dark-palette', 'seaborn-bright', 'seaborn-pastel', 'grayscale', 'seaborn-notebook', 'ggplot', 'seaborn-colorblind', 'seaborn-muted', 'seaborn', 'Solarize_Light2', 'seaborn-paper', 'bmh', 'tableau-colorblind10', 'seaborn-white', 'dark_background', 'seaborn-poster', 'seaborn-deep']
+```
+
+If we wanted to make sure our plots are colorblind friendly,
```
Comment (gvwilson) [inline py-rse/configuration.Rmd]: "accessible to people with color vision challenges"

File: py-rse/configuration.Rmd [hunk 5]
```diff
@@ -51,162 +53,361 @@ fonts:
...
+```
+
+Adding extra command line arguments is a reasonable solution
+if we only want to edit a small number of plot characteristics,
+but the number of new arguments could quickly get out of hand
+if we wanted to tweak other aspects of the plot
+(we saw in the `matplotlibrc` file that there are hundreds of parameters we could change). 
+For this particular example (i.e. matplotlib configuration),
+we ideally want a solution that can easily scale up to handling many parameters. 
+
```
Comment (gvwilson) [inline py-rse/configuration.Rmd]: remove extra blank line

File: py-rse/configuration.Rmd (hunk 1 above)
Comment (joelostblom) [reply py-rse/configuration.Rmd:80]: Wasn't it "run commands"? https://en.wikipedia.org/wiki/Run_commands Or is "resource control" even older?

Comment (DamienIrving) [reply py-rse/configuration.Rmd:80]: There were a few different takes on what `rc` has historically stood for at [this stack overflow question](https://stackoverflow.com/questions/37728087/rc-params-in-matplotlib-what-does-rc-stand-for), so I was reluctant to go beyond simply calling it "runtime configuration."

**Comment by DamienIrving** [comment]
Thanks, @gvwilson. It was fun - I learned a lot about matplotlib configuration!

**Pull request merged**